js/util/actor.js

```diff
@@ -28,6 +28,12 @@ Actor.prototype.receive = function(message) {
         id = data.id,
         callback;
 
+    // if we're on the main thread handling a message from the worker, make
+    // sure that we only handle messages directed at this actor's the map
+    // instance
+    if (data.targetMapId && this.mapId !== data.targetMapId)
+        return;
```

Comment:
Do you think there's a way you can make this code more self-documenting such that the comment is unnecessary?